src/nri/sampler.go

```diff
@@ -269,17 +269,35 @@ func pids(pids *biz.Pids) []entry {
 }
 
 func blkio(bio *biz.BlkIO) []entry {
-	return []entry{
-		metricIOTotalReadCount(bio.TotalReadCount),
-		metricIOTotalWriteCount(bio.TotalWriteCount),
-		metricIOTotalReadBytes(bio.TotalReadBytes),
-		metricIOTotalWriteBytes(bio.TotalWriteBytes),
-		metricIOTotalBytes(bio.TotalReadBytes + bio.TotalWriteBytes),
-		metricIOReadCountPerSecond(bio.TotalReadCount),
-		metricIOWriteCountPerSecond(bio.TotalWriteCount),
-		metricIOReadBytesPerSecond(bio.TotalReadBytes),
-		metricIOWriteBytesPerSecond(bio.TotalWriteBytes),
+	var entries []entry
+
+	if bio.TotalReadCount != nil {
+		entries = append(entries, metricIOTotalReadCount(bio.TotalReadCount), metricIOReadCountPerSecond(bio.TotalReadCount))
+	}
+	if bio.TotalWriteCount != nil {
+		entries = append(entries, metricIOTotalWriteCount(bio.TotalWriteCount), metricIOWriteCountPerSecond(bio.TotalWriteCount))
+	}
+	if bio.TotalReadBytes != nil {
+		entries = append(entries, metricIOTotalReadBytes(bio.TotalReadBytes), metricIOReadBytesPerSecond(bio.TotalReadBytes))
+	}
+	if bio.TotalWriteBytes != nil {
+		entries = append(entries, metricIOTotalWriteBytes(bio.TotalWriteBytes), metricIOWriteBytesPerSecond(bio.TotalWriteBytes))
 	}
+
+	var totalBytes *float64
+	if bio.TotalReadBytes != nil || bio.TotalWriteBytes != nil {
+		sum := 0.0
+		if bio.TotalReadBytes != nil {
+			sum += *bio.TotalReadBytes
+		}
+		if bio.TotalWriteBytes != nil {
+			sum += *bio.TotalWriteBytes
+		}
+		totalBytes = &sum
```

Comment:
I don't think this is needed right?

Reply:
just `entries = append(entries, metricIOTotalBytes(sum))` and perhaps replacing the name of the varialble `sum` for `totalBytes`

Reply:
Yes, we can simplify it by just using one scoped variable 👍

Reply:
Updated!